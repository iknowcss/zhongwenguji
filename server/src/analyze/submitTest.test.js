const express = require('express');
const supertest = require('supertest');
const submitTest = require('./submitTest');

describe('submitTest', () => {
  let body;

  beforeAll(() => {
    const app = express();
    app.post('/', submitTest());
    return supertest(app)
      .post('/')
      .send({
        testData: [
          { isTested: true, knownPercent: 100, range: [0, 250] },
          { isTested: false, knownPercent: NaN, range: [250, 500] },
          { isTested: true, knownPercent: 50, range: [500, 750] },
          { isTested: true, knownPercent: 20, range: [750, 1000] },
          { isTested: true, knownPercent: 30, range: [1000, 1250] },
          { isTested: true, knownPercent: 5, range: [1250, 1500] },
          { isTested: true, knownPercent: 0, range: [1500, 1750] },
          { isTested: true, knownPercent: 0, range: [1750, 2000] },
          { isTested: false, knownPercent: NaN, range: [2000, 2250] },
          { isTested: false, knownPercent: NaN, range: [2250, 2500] }
        ]
      })
      .expect(200)
      .then((result) => {
        body = result.body;
      });
  });

  it('calculates curve points', () => {
    const { curvePoints } = body;
    expect(curvePoints.slice(0, 5).map(([x]) => x))
      .toEqual([0, 250, 500, 750, 1000]);
    expect(curvePoints).toMatchSnapshot();
  });

  it('calculates sample points', () => {
    const { samplePoints } = body;
    expect(samplePoints).toEqual([
      [125, 100],
      [625, 50],
      [875, 20],
      [1125, 30],
      [1375, 5],
      [1625, 0],
      [1875, 0]
    ]);
  });

  it('known character range estimate', () => {
    const { knownEstimate, knownEstimateUncertainty } = body;
    expect({ knownEstimate, knownEstimateUncertainty }).toMatchSnapshot();
  });
});
