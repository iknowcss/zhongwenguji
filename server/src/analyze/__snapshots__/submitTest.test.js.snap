// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitTest calculates curve points 1`] = `
Array [
  Array [
    0,
    107.3692512060138,
  ],
  Array [
    250,
    85.86481560983094,
  ],
  Array [
    500,
    63.28538715426205,
  ],
  Array [
    750,
    41.71948555848205,
  ],
  Array [
    1000,
    23.161882781985035,
  ],
  Array [
    1250,
    9.329093457353348,
  ],
  Array [
    1500,
    1.5006031717680246,
  ],
  Array [
    1750,
    0,
  ],
  Array [
    2000,
    0,
  ],
  Array [
    2250,
    0,
  ],
  Array [
    2500,
    0,
  ],
]
`;

exports[`submitTest known character range estimate 1`] = `
Object {
  "knownEstimate": 700,
  "knownEstimateUncertainty": 50,
}
`;
